api/src/main/java/io/druid/data/input/impl/CSVParseSpec.java

```diff
@@ -35,26 +34,38 @@
 {
   private final String listDelimiter;
   private final List<String> columns;
+  private final boolean hasHeaderRow;
+  private final int skipHeaderRows;
 
   @JsonCreator
   public CSVParseSpec(
       @JsonProperty("timestampSpec") TimestampSpec timestampSpec,
       @JsonProperty("dimensionsSpec") DimensionsSpec dimensionsSpec,
       @JsonProperty("listDelimiter") String listDelimiter,
-      @JsonProperty("columns") List<String> columns
+      @JsonProperty("columns") List<String> columns,
```

Comment:
Just double checked this, an old-format JSON spec that doesn't have the new fields deserializes fine (gets hasHeaderRow=false, skipHeaderRows=0).

Looks like this does need a legacy constructor though.